packages/rest/src/__tests__/fixtures/controllers/note.controller.ts

```diff
@@ -0,0 +1,200 @@
+      },
+    })
+    note: Omit<Note, 'id'>,
+  ): Promise<Note> {
+    return Promise.resolve(groceryNote);
+  }
+
+  @get('/notes/count', {
+    responses: {
+      '200': {
+        description: 'Note model count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async count(
+    @param.query.object('where', getWhereSchemaFor(Note)) where?: Where<Note>,
+  ): Promise<Count> {
+    return Promise.resolve({count: 1});
+  }
+
+  @get('/notes', {
+    responses: {
+      '200': {
+        description: 'Array of Note model instances',
+        content: {
+          'application/json': {
+            schema: {
+              type: 'array',
+              items: getModelSchemaRef(Note, {includeRelations: true}),
+            },
+          },
+        },
+      },
+    },
+  })
+  async find(
+    @param.query.object('filter', getFilterSchemaFor(Note))
+    filter?: Filter<Note>,
+  ): Promise<Note[]> {
+    return Promise.resolve([groceryNote]);
+  }
+
+  @patch('/notes', {
+    responses: {
+      '200': {
+        description: 'Note PATCH success count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async updateAll(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(Note, {partial: true}),
+        },
+      },
+    })
+    note: Note,
+    @param.query.object('where', getWhereSchemaFor(Note)) where?: Where<Note>,
+  ): Promise<Count> {
+    return Promise.resolve({count: 1});
+  }
+
+  @get('/notes/{id}', {
+    responses: {
+      '200': {
+        description: 'Note model instance',
+        content: {
+          'application/json': {
+            schema: getModelSchemaRef(Note, {includeRelations: true}),
+          },
+        },
+      },
+    },
+  })
+  async findById(
+    @param.path.number('id') id: number,
+    @param.query.object('filter', getFilterSchemaFor(Note))
+    filter?: Filter<Note>,
+  ): Promise<Note> {
+    return Promise.resolve(groceryNote);
+  }
+
+  @patch('/notes/{id}', {
+    responses: {
+      '204': {
+        description: 'Note PATCH success',
+      },
+    },
+  })
+  async updateById(
+    @param.path.number('id') id: number,
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(Note, {partial: true}),
+        },
+      },
+    })
+    note: Note,
+  ): Promise<void> {
+    return Promise.resolve();
+  }
+
+  @put('/notes/{id}', {
+    responses: {
+      '204': {
+        description: 'Note PUT success',
+      },
+    },
+  })
+  async replaceById(
+    @param.path.number('id') id: number,
+    @requestBody() note: Note,
+  ): Promise<void> {
+    return Promise.resolve();
+  }
+
+  @del('/notes/{id}', {
+    responses: {
+      '204': {
+        description: 'Note DELETE success',
+      },
+    },
+  })
+  async deleteById(@param.path.number('id') id: number): Promise<void> {
+    return Promise.resolve();
+  }
```

Comment:
Do we really need all these CRUD controller methods for the test? Please keep only those methods that are relevant. I think that means either one method (to verify that both mixed-in and user-provided methods are recognized) or no methods at all (because we want to verify that mixed-in methods are added, we don't really care about user-provided methods).